### ISSUE TITLE
missing key prop on the config object.

### ISSUE THREAD

**Issue by baskint** [actions-on-google/dialogflow-name-psychic-nodejs#8]
Following the instructions, i was getting this:

```
actions-on-google/dialogflow-name-psychic-nodejs/functions
> eslint --fix "**/*.js"

✔  functions: Finished running predeploy script.
i  functions: ensuring necessary APIs are enabled...
✔  functions: all necessary APIs are enabled
i  functions: preparing functions directory for uploading...

Error: Error occurred while parsing your function triggers.

TypeError: Cannot read property 'key' of undefined
    at Object.<anonymous> (/Users/baskint/src/actions-on-google/dialogflow-name-psychic-nodejs/functions/index.js:25:25)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at /usr/local/lib/node_modules/firebase-tools/lib/triggerParser.js:18:11
    at Object.<anonymous> (/usr/local/lib/node_modules/firebase-tools/lib/triggerParser.js:38:3)
```
Upon inspection, if we change the line 25 of `functions/index.js` file to this, we get our API key back.
`  key: config.map.key`
After this change, I was able to deploy the firebase function.
```
  functions: Finished running predeploy script.
i  functions: ensuring necessary APIs are enabled...
✔  functions: all necessary APIs are enabled
i  functions: preparing functions directory for uploading...
i  functions: packaged functions (42.29 KB) for uploading
✔  functions: functions folder uploaded successfully
i  functions: creating function namePsychic...
✔  functions[namePsychic]: Successful create operation.
Function URL (namePsychic): https://***

✔  Deploy complete!
```
Now test simulator is crapped out, but that is across all the Google Action projects. Oh well.

**Comment by Fleker** [comment]
What command did you run to set the API key?

**Comment by baskint** [comment]
it is in step 7.III of the README.
``` 
firebase functions:config:set maps.key="<THE API KEY>"
```
did i misinterpret it? thanks

**Comment by baskint** [comment]
figured the error coming back on almost all the `Simulator` apps when `App information` is not starting. If i save all the information, the `Simulator` starts fine. Just letting you know. Thanks for listening (yet the samples never mention that)
```
TypeError: Cannot read property 'getAppInfo' of undefined
    at FBC_Hia (https://www.gstatic.com/_/aogconsole/_/js/k=aogconsole.a7t.en_US.vsJD8a0NFgk.O/m=a7t_console_main/am=GAzkAYAF/rt=j/d=1/rs=AGMOpTumD_114ru8pqqU1vA75tp8BxQcxw:1492:209)
    at new FBC_6o (https://www.gstatic.com/_/aogconsole/_/js/k=aogconsole.a7t.en_US.vsJD8a0NFgk.O/m=a7t_console_main/am=GAzkAYAF/rt=j/d=1/rs=AGMOpTumD_114ru8pqqU1vA75tp8BxQcxw:1446:494)
    at Object.instantiate (third_party/javascript/angular/v1_6/angular.min.js:40:67)
```

**Comment by Fleker** [comment]
Can you clarify what you mean and where this stacktrace comes from?

**Comment by baskint** [comment]
This above stack trace was coming from the Chrome Dev console, but I cannot seem to regenerate it. There could be a few minutes of delay when the app is generated and test simulator is ready. I had observed this error on the test simulator as it is invoked a few seconds after the app is created. 

I am seeing the list of intents, and I cannot make sure what the last entry should be. I deleted and restored the intents from the ZIP file as instructed. Still seeing the last entry.

![image](https://user-images.githubusercontent.com/211107/35808235-befff526-0a4a-11e8-9169-0514883dc851.png)
